pkg/controller/petset/identity_mappers.go

```diff
@@ -0,0 +1,202 @@
+package petset
+
+import (
+	"crypto/md5"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	api_pod "k8s.io/kubernetes/pkg/api/pod"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+const svcLocalSuffix = "svc.cluster.local"
+
+// identityMapper is an interface for assigning identities to a pet.
+// All existing identity mappers just append "-(index)" to the petset name to
```

Comment:
Why is it called "identity mapper"? It sounds like "unique name mapper" or "unique name manager"